lib/fastlane/plugin/wpmreleasetoolkit/versioning/formatters/derived_build_code_formatter.rb

```diff
@@ -19,15 +28,55 @@ class DerivedBuildCodeFormatter
         # @return [String] The formatted build code string.
         #
         def build_code(build_code = nil, version:)
-          format(
-            # 1 is appended to the beginning of the string in case there needs to be additional platforms or
-            # extensions that could then use a different digit prefix such as 2, etc.
-            '1%<major>.2i%<minor>.2i%<patch>.2i%<build_number>.2i',
+          result = format(
+            # The prefix is configurable to allow for additional platforms or
+            # extensions that could use a different digit prefix such as 2, etc.
+            '%<prefix>s%<major>.2i%<minor>.2i%<patch>.2i%<build_number>.2i',
+            prefix: @prefix,
             major: version.major,
             minor: version.minor,
             patch: version.patch,
             build_number: version.build_number
           )
+
+          # Only trim leading zeros when prefix is empty (to avoid removing explicit "0" prefix)
+          if @prefix.empty?
+            result.gsub(/^0+/, '')
+          else
+            result
+          end
+        end
+
+        private
+
+        # Validates that the prefix is a valid single digit (0-9) or empty string.
+        #
+        # @param [String] prefix The prefix to validate
+        #
+        # @raise [StandardError] If the prefix is invalid
+        #
+        def validate_prefix!(prefix)
+          prefix_str = prefix.to_s
+
+          # Allow empty string
+          return if prefix_str.empty?
+
+          # Check if it's longer than 1 character
+          if prefix_str.length > 1
+            UI.user_error!("Prefix must be a single digit or empty string, got: '#{prefix_str}' (length: #{prefix_str.length})")
+          end
+
+          # Check if it's a valid integer
+          begin
+            prefix_int = Integer(prefix_str)
+          rescue ArgumentError
+            UI.user_error!("Prefix must be an integer digit (0-9) or empty string, got: '#{prefix_str}'")
+          end
+
+          # Check if it's within valid range (0-9)
+          return if prefix_int.between?(0, 9)
+
+          UI.user_error!("Prefix must be a single digit (0-9), got: #{prefix_int}")
```

Comment:
I'm not sure how it'd be technically possible to get to that point in the code and fail that test, given we already ensured that `prefix_str` would not be a string longer than 1 and would be a valid `Integer` 😅 🙃